x-pack/legacy/plugins/apm/server/lib/jvms/index.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Setup } from '../helpers/setup_request';
+import { getJvmsProjection } from '../../../common/projections/jvms';
+import { mergeProjection } from '../../../common/projections/util/merge_projection';
+import {
+  METRIC_PROCESS_CPU_PERCENT,
+  METRIC_JAVA_THREAD_COUNT,
+  METRIC_JAVA_HEAP_MEMORY_USED,
+  METRIC_JAVA_NON_HEAP_MEMORY_USED
+} from '../../../common/elasticsearch_fieldnames';
+
+const getJvms = async ({
+  setup,
+  serviceName,
+  sortField,
+  sortDirection
```

Comment:
```suggestion
  sortDirection = 'asc'
```

Reply:
Default parameters don't work when a string is empty rather than null or undefined, so I wanted a `||` fallback. What do you think?

Reply:
> Default parameters don't work when a string is empty rather than null or undefined

Is that currently the case? I think most times when a value is set to `""` it is either a hack and should be avoided or it's a deliberate value in which case `||` will cause problems.

Reply:
`sort{Field,Direction}` is coming from an API endpoint, so it's hard to tell where exactly it's coming from. We could create a non-empty-string `io-ts` type as well. For now I'll just stick to your suggestion then.

Reply:
Okay, I thought it was values we passed in. I'm fine with leaving it as-is then.